### PR TITLE
fix: Watch mode not working on Windows

### LINKED ISSUE #25: Garment watch mode not working on Windows
### Description 

Example config :

```
"projects": {
    "dev-hub": {
      "path": "apps/dev-hub",
      "tasks": {
        "start": {
          "runner": "bin",
          "buildDependencies": {
            "task": "build",
            "watch": true
          },
```

When I make a change in a a file that is a dependency of `apps/dev-hub`,  the watcher should work on Windows.  Currently it is only working on Mac.

### PR DESCRIPTION
**Pull request by stephan-noel**
# Description

Fixes watch mode not working on windows.

Fixes #25 

## How Has This Been Tested?

Tried change in node_modules of the project where error was occurring and debugged the effects.

This can still be tested on a Mac to ensure nothing was broken.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have added tests to cover my changes
- [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: core/garment/src/garment.ts [hunk 1]
```diff
@@ -919,7 +919,7 @@ async function garmentFromWorkspace(
 
         if (
           subscription.type === 'glob' &&
-          event.path.startsWith(subscription.input.rootDir)
+          event.path.startsWith(Path.normalize(subscription.input.rootDir))
```
Comment (stephan-noel) [inline core/garment/src/garment.ts]: @beshanoe I'm not really comfortable with this change.  Though it does work, it adds extra file input-level subscriptions to `allSubscriptionsCached`:

Before fix:

```
{
    "type": "file",
    "level": "input",
    "path": "C:\\Users\\stephan.noel\\farfetch-repos\\ui-box\\babel.config.js",
    "targetPath": "C:/Users/stephan.noel/farfetch-repos/ui-box/packages/accordion/src/context/focus-provider/FocusProvider.js"
}, {
    "type": "glob",
    "level": "input",
    "input": {
        "rootDir": "C:/Users/stephan.noel/farfetch-repos/ui-box/packages/typography/src",
        "include": ["**/*.js?(x)"],
        "exclude": ["**/*.test.js", "**/__fixtures__", "**/__mocks__"]
    }
}, {
    "type": "file",
    "level": "input",
    "path": "C:\\Users\\stephan.noel\\farfetch-repos\\ui-box\\babel.config.js",
    "targetPath": "C:/Users/stephan.noel/farfetch-repos/ui-box/packages/typography/src/index.js"
}
```

After fix:

```
{
    "type": "file",
    "level": "input",
    "path": "C:\\Users\\stephan.noel\\farfetch-repos\\ui-box\\babel.config.js",
    "targetPath": "C:/Users/stephan.noel/farfetch-repos/ui-box/packages/accordion/src/context/focus-provider/FocusProvider.js"
}, {
    "type": "file",
    "level": "input",
    "path": "C:\\Users\\stephan.noel\\farfetch-repos\\ui-box\\babel.config.js",
    "targetPath": "C:\\Users\\stephan.noel\\farfetch-repos\\ui-box\\packages\\accordion\\src\\components\\AccordionButton.js"
}, {
    "type": "glob",
    "level": "input",
    "input": {
        "rootDir": "C:/Users/stephan.noel/farfetch-repos/ui-box/packages/typography/src",
        "include": ["**/*.js?(x)"],
        "exclude": ["**/*.test.js", "**/__fixtures__", "**/__mocks__"]
    }
}, {
    "type": "file",
    "level": "input",
    "path": "C:\\Users\\stephan.noel\\farfetch-repos\\ui-box\\babel.config.js",
    "targetPath": "C:/Users/stephan.noel/farfetch-repos/ui-box/packages/typography/src/index.js"
}
```
We could try the approach of changing what's stored in the cache itself, but this would include changing the createdInput, rootDir (probably in getActionGraph) and targetPath to be OS-specific.  Not sure if the TypeScript compiler approach of dealing with only one kind of path internally is a good strategy?

image (27)

TLDR;

This "fix" should work for now, but it has some unwanted effects.  Taking the approach of platform-specific paths in the cache itself may imply changes in a lot of places.  Which do you think is preferable?

Comment (beshanoe) [reply core/garment/src/garment.ts]: I didnt understand how does it changes allSubscriptionCached? It seems to only affect changesBySubscriptions. We can also probably normalize the path before adding it to the subscription? 🤔 
Is the path unix-style inside the watcher event?

Comment (stephan-noel) [reply core/garment/src/garment.ts]: Changed the `changesBySubscription` entry to normalize as suggested, but using `normalizePath`.  It's now working without extra entries in `allSubscriptionsCached`.

It is failing in build due to an outdated github action.  I can try looking into it and after it should be ready to merge.

Comment (beshanoe) [reply core/garment/src/garment.ts]: can we use `normalizePath` on this line too? Just to make it look unified. Or they work differently in this case?

Comment (stephan-noel) [reply core/garment/src/garment.ts]: Sure thing.  They basically do the opposite, so I can just use `normalize-path` on `event.path` instead of using `Path.normalize` on `subscription.input.rootDir`.  Changed, what do you think?

Comment (beshanoe) [reply core/garment/src/garment.ts]: Looks good to me
